apps/sms/test/unit/sms_test.js

```diff
@@ -527,43 +528,44 @@ suite('SMS App Unit-Test', function() {
           type: 'sms',
           channel: 'sms'
         };
-        ThreadUI.appendMessage(incomingMessage);
 
-        // new checkbox should have been added
-        checkboxes = ThreadUI.container.querySelectorAll(
-          'input[type=checkbox]'
-        );
-        assert.equal(checkboxes.length, 6);
-        assert.equal(checkboxes[2].checked, true);
-        assert.equal(checkboxes[5].checked, false);
-
-        // Select-Deselect all should both be enabled
-        assert.isFalse(
-          checkUncheckAllButton.hasAttribute('disabled'),
-          'Check-Uncheck all enabled'
-        );
+        ThreadUI.appendMessage(incomingMessage).then(() => {
+          // new checkbox should have been added
+          checkboxes = ThreadUI.container.querySelectorAll(
+            'input[type=checkbox]'
+          );
+          assert.equal(checkboxes.length, 6);
+          assert.equal(checkboxes[2].checked, true);
+          assert.equal(checkboxes[5].checked, false);
+
+          // Select-Deselect all should both be enabled
+          assert.isFalse(
+            checkUncheckAllButton.hasAttribute('disabled'),
+            'Check-Uncheck all enabled'
+          );
 
-        // now delete the selected messages...
-        this.sinon.stub(MessageManager, 'deleteMessages').yields();
+          // now delete the selected messages...
+          this.sinon.stub(MessageManager, 'deleteMessages').yields();
 
-        var getMessageReq = {
-          result: incomingMessage
-        };
-        this.sinon.stub(MessageManager, 'getMessage').returns(getMessageReq);
-
-        ThreadUI.delete().then(() => {
-          getMessageReq.onsuccess();
-          sinon.assert.calledOnce(MessageManager.deleteMessages);
-          assert.equal(MessageManager.deleteMessages.args[0][0].length, 5);
-          assert.equal(ThreadUI.container.querySelectorAll('li').length, 1,
-            'correct number of Thread li');
-          assert.equal(
-            ThreadUI.container.querySelector('#message-9999 p').textContent,
-            'Recibidas!');
-        }).then(done, done);
+          var getMessageReq = {
+            result: incomingMessage
+          };
+          this.sinon.stub(MessageManager, 'getMessage').returns(getMessageReq);
+
+          ThreadUI.delete().then(() => {
+            getMessageReq.onsuccess();
+            sinon.assert.calledOnce(MessageManager.deleteMessages);
+            assert.equal(MessageManager.deleteMessages.args[0][0].length, 5);
+            assert.equal(ThreadUI.container.querySelectorAll('li').length, 1,
+              'correct number of Thread li');
+            assert.equal(
+              ThreadUI.container.querySelector('#message-9999 p').textContent,
+              'Recibidas!');
+          }).then(done, done);
```

Comment:
I think we should move ```.then(done, done);``` to the top most promise (ThreadUI.appendMessage().then(done, done)), otherwise if we have failed assert above ```ThreadUI.delete```, test will fail with timeout.

Maybe like 
```
ThreadUI.appendMessage(() => {
  ....
  return ThreadUI.delete().then(() => {
    ...asserts...
  });
}).then(done, done);

or 

ThreadUI.appendMessage(() => {
  ....
  return ThreadUI.delete();
}).then(() => {
  ...asserts...
});then(done, done);
```

What do you think?